### ISSUE TITLE
Image.to_rgba8() fails

### ISSUE THREAD

**Issue by tcmal** [ggez/ggez#371]
Code:

```
extern crate ggez;

use std::env;
use std::path;

use ggez::{conf, event, Context};
use ggez::graphics::Image;

fn main() {
    let c = conf::Conf::new();
    let ctx = &mut Context::load_from_conf("test", "me", c).unwrap();


    if let Ok(manifest_dir) = env::var("CARGO_MANIFEST_DIR") {
        let mut path = path::PathBuf::from(manifest_dir);
        path.push("resources");
        ctx.filesystem.mount(&path, true);
    }

	let i = Image::new(ctx, "/spritesheet.png").unwrap().to_rgba8(ctx).unwrap();
    
	println!("{:?}", i);
}
```

Output:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: VideoError("Memory copy error: NoSrcBindFlag")', libcore/result.rs:945:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

It seems like the [right flag](https://docs.rs/gfx/0.17.1/gfx/memory/struct.Bind.html) could be inserted but I'm not sure if that's correct as I have no experience with OpenGL/low-level graphics.

**Comment by icefoxen** [comment]
Ooooh.  That shouldn't be happening.

Thank you!  I will try to reproduce it.  Should be an easy fix if so.

For reference, what OS and graphics card are you using?

**Comment by tcmal** [comment]
I'll make a PR for

**Comment by icefoxen** [comment]
I'm working on it but it's harder than I expected cause it runs into a couple of bumps in how gfx handles typed textures. (It's more difficult than it should be to set bind flags on textures AND load data into them.) I think I have a workaround but I need to test it more thoroughly. Sorry for the delay.

**Comment by icefoxen** [comment]
Should be fixed in  f7f6b48!  Sorry for the delay.  Please test and make sure it works for you.

**Comment by tcmal** [comment]
Yep, that works.
